src/main/java/com/hotels/beeju/BeejuJUnitRule.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (C) 2015-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.File;
+import java.util.UUID;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.conf.HiveConf.ConfVars;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.thrift.TException;
+import org.hsqldb.jdbc.JDBCDriver;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.annotations.VisibleForTesting;
+
+abstract class BeejuJUnitRule extends ExternalResource {
+
+  public static final String HSQLDB_USER = "user";
+  public static final String HSQLDB_PASSWORD = "password";
+
+  @VisibleForTesting
+  final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  protected final HiveConf conf = new HiveConf();
+  private final String databaseName;
+  private final String connectionURL;
+  private final String driverClassName;
+  private File metastoreLocation;
+
+  public BeejuJUnitRule(String databaseName) {
+    checkNotNull(databaseName, "databaseName is required");
+    this.databaseName = databaseName;
+    driverClassName = JDBCDriver.class.getName();
+    conf.setBoolean("hcatalog.hive.client.cache.disabled", true);
+    connectionURL = "jdbc:hsqldb:mem:" + UUID.randomUUID();
+    conf.setVar(ConfVars.METASTORECONNECTURLKEY, connectionURL);
+    conf.setVar(ConfVars.METASTORE_CONNECTION_DRIVER, driverClassName);
+    conf.setVar(ConfVars.METASTORE_CONNECTION_USER_NAME, HSQLDB_USER);
+    conf.setVar(ConfVars.METASTOREPWD, HSQLDB_PASSWORD);
+    conf.setBoolVar(ConfVars.HMSHANDLERFORCERELOADCONF, true);
+    // Hive 2.x compatibility
+    conf.setBoolean("datanucleus.schema.autoCreateAll", true);
+  }
+
+  /**
+   * Initialise the warehouse path.
+   * <p>
+   * This method can be overridden to provide additional initialisations.
+   *
+   * @throws Throwable If the initialisation fails.
+   */
+  protected void init() throws Throwable {
+    metastoreLocation = temporaryFolder.newFolder("metastore");
+    conf.setVar(ConfVars.METASTOREWAREHOUSE, metastoreLocation.getAbsolutePath());
+  }
+
+  /**
+   * Implement method to prepare the rule for a new test.
+   * <p>
+   * This method is called after the rule is initialised.
```

Comment:
Either add `</p>` or don't bother with a paragraph at all and just have two sentences next to each other.